src/utils.js

```diff
@@ -71,6 +71,28 @@ const utils = {
     return immutableData.every((item) => !item || item.isEmpty());
   },
 
+  flattenMap(data) {
+    return data.reduce(
+      (flattened, section, key) => flattened.set(key, section).merge(section),
+      Immutable.OrderedMap().asMutable(),
```

Comment:
I think I understand but please clarify -- is `asMutable` being used here for performance reasons, so we don't have to return a new Immutable object during each reduction step?

Reply:
Yes exactly, it's a little trick I learned after months of using immutablejs. Within our `flattenMap` function, it's safe to make the collection mutable while we build it as long as we call `asImmutable()` on it before we return it.